### PR TITLE
Change flanking to 1.0, and scale HP and death explosions to compensate

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
As discussed on discord, the engine default flanking damage will be changed from min=0.9, max=1.9 to min=1.0, max=2.0.

Setting flanking max=2.0 instead of 2.11 will result in an approximately 5% nerf to overall bonus damage from flanking. 

All health values scaled by a factor of 10/9, (111%), to compensate for all attacks no longer being reduced by a factor of 9/10, (90%), due to the flanking system. Scaled values were rounded using the rule:
* If the first significant digit is a 1, round to 3 significant figures.
* If the first significant digit is 2, 3, or 4, round the third significant figure to a 0 or 5 (so 3333 would be rounded to 3350).
* If the first significant digit is 5 or more, round to 2 significant figures.
This rounding rule constrains all health variance from rounding (compared to perfect 10/9 scaled values) to +-1%.

Scaling health (instead of damage) to compensate for flanking changes causes the least disruption. 
Smaller percentage changes due to rounding (due to larger base values, and greater than 1 scaling factor).
Already existing health variance due to XP system. 
And systems unaffected by flanking such as damage to map feature, and damage to shields are unaffected.

Unit death explosion damage was never scaled by the flaking system, so these death explosion damages have also been increased by a factor of 10/9 and rounded to compensate for the increased health of all units.

Wrecks left by units will be 11.1% more durable after this change, as unit base health is being increased, but the flanking multiplier increase is not propagated to damage dealt to features.

Autoheal values may need to be looked at after this change, as this health increase nerfs all autohealing by 11%.

### CODE REVIEW

File: gamedata/alldefs_post.lua [hunk 1]
```diff
@@ -101,6 +101,8 @@ function UnitDef_Post(name, uDef)
 			uDef.flankingbonusmode = 2
 		end
 	end
+	uDef.flankingbonusmin = 1.0 -- change from default engine value of 0.9
+	uDef.flankingbonusmax = 2.0 -- change from default engine value of 1.9
```
Comment (WatchTheFort) [inline gamedata/alldefs_post.lua:105]: These two lines are no-op due to the next two lines overwriting the value. Merge the branch https://github.com/beyond-all-reason/Beyond-All-Reason/tree/flanking-rework, which sets the default values in modrules.lua, and removes the experimental flanking modoptions. These two lines can then be removed from alldefs_post.lua.

**Review by WatchTheFort — CHANGES_REQUESTED** [review]

File: gamedata/alldefs_post.lua (hunk 1 above)
Comment (KyleAnthonyShepherd) [reply gamedata/alldefs_post.lua]: flanking-rework should now be merged

**Review by WatchTheFort — APPROVED** [review]

**Comment by Beherith** [comment]
@Zecrus2 , @PtaQQ  please review and decide how this should go forward.

**Comment by WatchTheFort** [comment]
Whatever happened with this? It got closed, but it didn't get merged.

**Comment by sprunk** [comment]
probably means it got rejected

**Comment by sprunk** [comment]
oops

**Comment by WatchTheFort** [comment]
@Beherith?

**Comment by Beherith** [comment]
Dunno, im not the authority on this one, see above. I might have accidentally closed it tho.

**Pull request merged**